roles/integration-test/Cargo.toml

```diff
@@ -0,0 +1,9 @@
+[package]
+name = "integration-test"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+
+[[bin]]
```

Comment:
Apparently we need to have either `bin` or `main` or `lib` somewhere in a crate, so although this is not used, it is needed.

Reply:
I pulled your branch and:
- deleted `src/bin`
- removed the `[[bin]]` from `Cargo.toml`
- did `cargo test`

everything worked fine

Reply:
What if you run `cargo build` in `roles` workspace?

Reply:
built without issues

Reply:
thats weird.. I see this error 
```
Caused by:
  no targets specified in the manifest
  either src/lib.rs, src/main.rs, a [lib] section, or [[bin]] section must be present
```

Reply:
I honestly dont think its a huge deal, we can just add a comment at the top of the file explaining why its there..

Reply:
> thats weird.. I see this error
> 
> ```
> Caused by:
>   no targets specified in the manifest
>   either src/lib.rs, src/main.rs, a [lib] section, or [[bin]] section must be present
> ```

I'm getting the same error, with `rustc version = 1.75.0`

Reply:
here's a structure that IMO feels a bit more coherent:
```
├── Cargo.toml
├── src
│   └── lib.rs
└── tests
    └── pool_integration.rs
```

we could just move the contents under `tests/common/mod.rs` to `src/lib.rs`, which essentially turns this crate into a "library for integration tests"

Reply:
I set my `rustc` to 1.75.0 and still cannot reproduce this build error

but anyways, I agree with the suggestion provided here: https://github.com/stratum-mining/stratum/pull/1155/files#r1745253054

Reply:
Ill resolve this and we can discuss in https://github.com/stratum-mining/stratum/pull/1155